src/main/java/edu/escuelaing/ieti/matchmaking/persistence/impl/InMemoryUserRepository.java

```diff
@@ -0,0 +1,57 @@
+package edu.escuelaing.ieti.matchmaking.persistence.impl;
+
+import edu.escuelaing.ieti.matchmaking.model.User;
+import edu.escuelaing.ieti.matchmaking.persistence.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class InMemoryUserRepository implements UserRepository {
+
+    private static Map<String, User> userMap = new ConcurrentHashMap<>();
+
+    @Override
+    public User create(User user) {
+        String userId = user.getUserId();
+        if (!userMap.containsKey(userId)){
+            userMap.put(userId, user);
+            return user;
+        }
+        return null;
```

Comment:
Consider throwing an exception or returning an optional to avoid a null pointer exception